src/Microsoft.AspNet.Mvc.Core/ParameterBinding/ModelBindingHelper.cs

```diff
@@ -162,6 +167,8 @@ public static class ModelBindingHelper
 
             if (await modelBinder.BindModelAsync(modelBindingContext))
             {
+                var modelValidationContext = new ModelValidationContext(modelBindingContext, modelMetadata);
+                objectModelValidator.Validate(modelValidationContext, prefix);
```

Comment:
> A property like RootPrefix does not make a lot of sense inside the ModelValidatorContext primarily because its not needed at every level.

The purpose of the context is to gather the set of arguments needed for the validator. It's not a contract for property validators or for how the validator should be implemented